Sources/TestingMacros/Support/ClosureCaptureListParsing.swift

```diff
@@ -76,18 +77,71 @@ struct CapturedValueInfo {
         // Copying self.
         self.type = typeNameOfLexicalContext
       } else {
-        context.diagnose(.typeOfCaptureIsAmbiguous(capture, initializedWith: initializer))
+        // Handle literals. Any other types are ambiguous.
+        switch self.expression.kind {
+        case .integerLiteralExpr:
+          self.type = TypeSyntax(IdentifierTypeSyntax(name: .identifier("IntegerLiteralType")))
+        case .floatLiteralExpr:
+          self.type = TypeSyntax(IdentifierTypeSyntax(name: .identifier("FloatLiteralType")))
+        case .booleanLiteralExpr:
+          self.type = TypeSyntax(IdentifierTypeSyntax(name: .identifier("BooleanLiteralType")))
+        case .stringLiteralExpr, .simpleStringLiteralExpr:
+          self.type = TypeSyntax(IdentifierTypeSyntax(name: .identifier("StringLiteralType")))
+        default:
+          context.diagnose(.typeOfCaptureIsAmbiguous(capture, initializedWith: initializer))
+        }
       }
 
     } else if capture.name.tokenKind == .keyword(.self),
               let typeNameOfLexicalContext {
       // Capturing self.
       self.expression = "self"
       self.type = typeNameOfLexicalContext
-
+    } else if let parameterType = Self._findTypeOfParameter(named: capture.name, in: lexicalContext) {
+      self.expression = ExprSyntax(DeclReferenceExprSyntax(baseName: capture.name.trimmed))
+      self.type = parameterType
     } else {
       // Not enough contextual information to derive the type here.
       context.diagnose(.typeOfCaptureIsAmbiguous(capture))
     }
   }
+
+  /// Find a function or closure parameter in the given lexical context with a
+  /// given name and return its type.
+  ///
+  /// - Parameters:
+  ///   - parameterName: The name of the parameter of interest.
+  ///   - lexicalContext: The lexical context to examine.
+  ///
+  /// - Returns: The Swift type of first parameter found whose name matches, or
+  ///   `nil` if none was found. The lexical context is searched in the order
+  ///   provided which, by default, starts with the innermost scope.
+  private static func _findTypeOfParameter(named parameterName: TokenSyntax, in lexicalContext: [Syntax]) -> TypeSyntax? {
+    for lexicalContext in lexicalContext {
+      var parameterType: TypeSyntax?
+      if let functionDecl = lexicalContext.as(FunctionDeclSyntax.self) {
+        parameterType = functionDecl.signature.parameterClause.parameters
+          .first { ($0.secondName ?? $0.firstName).tokenKind == parameterName.tokenKind }
+          .map(\.type)
```

Comment:
Does this need to contend with keywords like `borrowing` on the parameter type if present, or does it already? And could that be validated in a test either way?